cmd/snap/cmd_info.go

```diff
@@ -260,14 +260,30 @@ func (iw *infoWriter) maybePrintTrackingChannel() {
 	fmt.Fprintf(iw, "tracking:\t%s\n", iw.localSnap.TrackingChannel)
 }
 
-func (iw *infoWriter) maybePrintInstallDate() {
+func (iw *infoWriter) maybePrintRefreshInfo() {
 	if iw.localSnap == nil {
 		return
 	}
-	if iw.localSnap.InstallDate.IsZero() {
-		return
+
+	if !iw.localSnap.InstallDate.IsZero() {
+		fmt.Fprintf(iw, "refresh-date:\t%s\n", iw.fmtTime(iw.localSnap.InstallDate))
+	}
+
+	maybePrint := func(key string, hold time.Time) {
+		if hold.IsZero() {
+			return
+		}
+
+		longTime := timeNow().Add(100 * 365 * 24 * time.Hour)
+		if hold.After(longTime) {
+			fmt.Fprintf(iw, "%s:\tin a very long time\n", key)
+		} else {
+			fmt.Fprintf(iw, "%s:\t%s\n", key, iw.fmtTime(hold))
+		}
 	}
-	fmt.Fprintf(iw, "refresh-date:\t%s\n", iw.fmtTime(iw.localSnap.InstallDate))
+
+	maybePrint("user hold expires", iw.localSnap.UserHold)
+	maybePrint("gating hold expires", iw.localSnap.GatingHold)
```

Comment:
we traditionally never have spaces in the field names, so these should really be just:

```
hold:
hold-by-gating:
```

Reply:
Yeah I was unsure about this one, I changed it a bit so it would be more readable than `hold: in 2 days, at 14:58 GMT`. I'll change it. Should it still be an informal time or a precise timestamp (so it's useful for scripts)?

Reply:
iw.fmtTime respects --abs-time I think, so that solves your question?

Reply:
It does, thanks!